### PR TITLE
Reduce overhead in LoadBalancerSubsetInfoImpl by fixing alignment

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Reduce overhead in LoadBalancerSubsetInfoImpl by fixing alignment
Additional Description: Moves many of the smaller members of the class to the end to save small amounts of memory
Risk Level: Low

### CODE REVIEW

File: source/common/upstream/load_balancer_impl.h [hunk 1]
```diff
@@ -806,17 +806,17 @@ class LoadBalancerSubsetInfoImpl : public LoadBalancerSubsetInfo {
...
-      metadata_fallback_policy_;
   const ProtobufWkt::Struct default_subset_;
   std::vector<SubsetSelectorPtr> subset_selectors_;
-  const bool locality_weight_aware_;
-  const bool scale_locality_weight_;
-  const bool panic_mode_any_;
-  const bool list_as_any_;
+  const envoy::config::cluster::v3::Cluster::LbSubsetConfig::LbSubsetMetadataFallbackPolicy
+      metadata_fallback_policy_;
+  const bool enabled_ : 1;
```
Comment (adisuissa) [inline source/common/upstream/load_balancer_impl.h:815]: nit: please add a small note about the coalescing of these data members at the end (something like [this](https://github.com/envoyproxy/envoy/blob/1ac0a5948ca317fd418777dab2ffe27935c0a276/source/common/router/config_impl.h#L342)).

File: source/common/upstream/load_balancer_impl.h [hunk 2]
```diff
@@ -806,17 +806,17 @@ class LoadBalancerSubsetInfoImpl : public LoadBalancerSubsetInfo {
...
       fallback_policy_;
-  const envoy::config::cluster::v3::Cluster::LbSubsetConfig::LbSubsetMetadataFallbackPolicy
-      metadata_fallback_policy_;
   const ProtobufWkt::Struct default_subset_;
   std::vector<SubsetSelectorPtr> subset_selectors_;
-  const bool locality_weight_aware_;
-  const bool scale_locality_weight_;
-  const bool panic_mode_any_;
-  const bool list_as_any_;
+  const envoy::config::cluster::v3::Cluster::LbSubsetConfig::LbSubsetMetadataFallbackPolicy
```
Comment (adisuissa) [inline source/common/upstream/load_balancer_impl.h:813]: I see this field moved but not `fallback_policy_`, why?
Is there added benefit by moving these enum types?

**Review by adisuissa — COMMENTED** [review]
Thanks!
Left a nit and a question.
Can you also update the PR comment presenting the diffs in the class size?

File: source/common/upstream/load_balancer_impl.h (hunk 2 above)
Comment (DiazAlan) [reply source/common/upstream/load_balancer_impl.h:814]: Good catch, I just forgot to pull the other enum down. Technically, moving it does doesn't save memory but I feel it'll be better for consistency to move them down anyway.

File: source/common/upstream/load_balancer_impl.h (hunk 1 above)
Comment (DiazAlan) [reply source/common/upstream/load_balancer_impl.h:816]: Done

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

/assign-from @envoyproxy/senior-maintainers

**Review by snowp — APPROVED** [review]
Thanks!

**Pull request merged**